src/plugins/Miner.cpp

```diff
@@ -73,6 +73,75 @@ void CallDownMULE() {
     }
 }
 
+const Expansion* GetBestMiningExpansionNear(const sc2::Unit* unit_) {
+    if (!unit_)
+        return nullptr;
+
+    sc2::Point2D worker_loc = unit_->pos;
+    float target_dist = std::numeric_limits<float>::max();
+    const Expansion* target = nullptr;
+
+    std::vector<const Expansion*> SaturatedExpansions;
+
+    // Find closest unsaturated expansion
+    for (auto& i : gHub->GetExpansions()) {
+        if (i.owner != Owner::SELF)
+            continue;
+
+        const sc2::Unit* th = gAPI->observer().GetUnit(i.town_hall_tag);
+        if (th->build_progress < 1.0f)
+            continue;
```

Comment:
I think the good way to avoid such filtering is to assume that every region without finished downhill is contested.

Reply:
Also, we seems that we need to create a global constant for `1.0f` value as it is used in many places now.

Reply:
I think for now CONTESTED should only mean the army should clear away (possible) enemies so the worker can start building the townhall (and army remain there until finished, or new CONTESTED region appears, or enemy attacks elsewhere).

I think any refactor should be done in 'Secure/Defend new expansion' issue (#53) because it will redo and refine many of those things anyway.
(perhaps a new Expansion variable, like 'signal' for an ATTACK/DEFEND signal to tell army what to do)

Reply:
I think global variables are only useful if the value could possibly change, so it can be updated in just one place. But 1.0f will always mean "fully built", and it is already easily understandable.